routes/course.js

```diff
@@ -112,6 +112,14 @@ router.post("/api/courses", auth, async (req, res) => {
           message: "Invalid combination of plan_id and distribution_ids.",
         });
     });
+    const retrievedCourses = await Courses.findByPlanId(course.plan_id); 
+    for (const existingCourse of retrievedCourses) {
+      if(existingCourse.number === course.number && existingCourse.term === course.term && existingCourse.year === course.year && existingCourse.title === course.title){
+        return errorHandler(res, 400, { 
+          message: "Cannot take same course multiple times in the same semester",
```

Comment:
So in frontend, how will you differentiate between this particular error and other types of error handled in line 142 `errorHandler(res, 400, err);`? 

Are you making sure that the toast only displays this particular message? Asking because in both cases you are returning status code 400.

Reply:
how about you reserve 400 for the `Cannot take same course multiple times in the same semester` and return 500 otherwise? and in the frontend, you simply check for status code. 

In this case, you may have to update the jest tests as you're changing status code from 400 to 500.

Reply:
Just made the change!